### ISSUE TITLE
reproducibility problem

### ISSUE THREAD

**Issue by Hikizzz** [zjunlp/MKG_Analogy#8]
Hi,thanks for this work!
I‘m trying to reproduce the results on my PC(ubuntu18,torch2.0.1),but it seems to be strange.Taking hits@10 on mkgformer for example:
- pretrain and finetune: 0.3289
- finetune based on yout checkpoint "pretrain_mkgformer.ckpt": 0.4001
- only test your checkpoint "ft_mart_mkgformer.ckpt":  0.3789
- in your paper: 0.408
all Hyperparameters are the same as your codes except the patience for earlyStopping increase from 5 to 10.
Any idea why is it?

**Comment by flow3rdown** [comment]
Thanks for your attention!

I will check the results and get back to you soon.

**Comment by flow3rdown** [comment]
Hi,

We find that setting `g_ea=0` and `g_ae=1`(equation 4 in paper) helps`ft_mart_mkgformer.ckpt` most, so you can modify this part ([link](https://github.com/zjunlp/MKG_Analogy/blob/a261ea5acfcf76cc18254c301f7d24d35db93d29/MarT/data/data_module.py#L168-L171)) to the following code and do test:
```python
# features['sep_idx'] = torch.tensor(sep_idx)
features['attention_mask'] = features['attention_mask'].unsqueeze(1).expand([features['input_ids'].size(0), features['input_ids'].size(1), features['input_ids'].size(1)]).clone()
for i, idx in enumerate(sep_idx):
    features['attention_mask'][i, :idx[2], idx[2]:] = 0
```
But fine-tuning these two gates can approximate this result, as you said, fine-tuning can reach  0.4001.

**Comment by Hikizzz** [comment]
Thanks for your reply.
I've tried the above methods,test more models and read "Issue#9".Maybe I should try more Hyperparameters.
To summarize my experiments:
- the above method did works (hits@10: 0.3289 to 0.3319)but still worse than the results in the paper(0.408).I have summarized three questions.

-  In my enviroments,your checkpoints often perform poorly even only half such as "ft_mart_mkgformer.ckpt" ,"ft_mart_mkgformer.ckpt " and "pretrain_vilt.ckpt"
What's the possible reason?Hardware，library version，datasets or others?

and there are maybe some bugs:
- In "encode_images_data.py", when processor_name = 'vilt', "entity2visual = torch.cat(entity2visual, dim=0)" may cause out of system memory(24GB). 
- in "modeling_vilt.py" line 844, when pretraining, " sep_idx += img_length" turns into "None += int"
-  in "modeling_vilt.py" line 183,"select" and "patch_index" seems to be stored on different device.
- some scripts and checkpoints seem to be missing.
